File: Stage27/Python/Problem2606.py
import sys
from collections import deque

input = sys.stdin.readline

def bfs(start, graph, visited):
    queue = deque([start])
    visited[start] = True
    count = 0
    
    while queue:
        node = queue.popleft()
        
        for next_node in graph[node]:
            if not visited[next_node]:
                visited[next_node] = True
                queue.append(next_node)
                count += 1
    
    return count

def main():
    com = int(input())
    line = int(input())
    
    graph = [[] for _ in range(com + 1)]
    visited = [False] * (com + 1)
    
    for _ in range(line):
        a, b = map(int, input().split())
        graph[a].append(b)
        graph[b].append(a)
    
    print(bfs(1, graph, visited))

if __name__ == "__main__":
    main()

File: Stage27/Java/Problem2606.java
package Stage27.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class Problem2606 {
    static int com, line;
    static ArrayList<Integer>[] graph;
    static boolean[] visited;
    static int count = 0;

    static void dfs(int start) {
        visited[start] = true;

        for (int next : graph[start]) {
            if (!visited[next]) {
                dfs(next);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        com = Integer.parseInt(br.readLine());
        line = Integer.parseInt(br.readLine());
        
        graph = new ArrayList[com + 1];
        visited = new boolean[com + 1];

        for (int i = 1; i <= com; i++) {
            graph[i] = new ArrayList<>();
        }

        for (int i = 0; i < line; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a].add(b);
            graph[b].add(a);
        }

        dfs(1);

        for (int i = 2; i <= com; i++) {
            if (visited[i]) count++;
        }

        System.out.println(count);
        br.close();
    }
}
